src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -136,11 +136,7 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     proxy_cert = proxy_cert.replace('\\', r'\\\\')
 
     # Set preview client if latest preview properties are provided.
-    if enable_private_link is not None or distribution_version is not None or azure_hybrid_benefit is not None:
-        client = cf_connected_cluster_prev_2023_11_01(cmd.cli_ctx, None)
-
-    # Set preview client if the connection-type is provided. (TODO: To test whether overriding the client factory to 2024 will retain the 2023 private link feature as in the line above)
-    if connection_type is not None and connection_type == "gateway" or enable_workload_identity or enable_oidc_issuer:
+    if enable_private_link is not None or distribution_version is not None or azure_hybrid_benefit is not None or enable_workload_identity or enable_oidc_issuer:
```

Comment:
shouldnt we use the 2024-07-01 api only when workload identity is enabled in this case?

Reply:
Did we test that the enable_private_link and azure-hybrid-benefit works fine with this new api? If yes, then this is fine.

Reply:
yes I tested it and those properties work fine with the new api